mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/plan/synchronization/IndexingPlanSynchronizationStrategy.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.plan.synchronization;
+
+import org.hibernate.search.mapper.pojo.work.SearchIndexingPlanExecutionReport;
+import org.hibernate.search.util.common.annotation.Incubating;
+
+/**
+ * Determines how the thread will block upon committing a transaction
+ * where indexed entities were modified.
+ *
+ * {@code SearchSession#indexingPlanSynchronizationStrategy(IndexingPlanSynchronizationStrategy)}
+ */
+@Incubating
+public interface IndexingPlanSynchronizationStrategy<E, R extends SearchIndexingPlanExecutionReport<E>, C extends IndexingPlanSynchronizationStrategyConfigurationContext<E, R>> {
+
+	void apply(C context);
+
+}
```

Comment:
IMO this is either not enough, or too much.

If we're going to have a sub-interface for each mapper, then it's too much and there's little value in keeping this one, since it's so simple. We could just remove it, declare the method directly in each mapper's interface, and use static methods to share the implementation of built-in strategies across mappers.

But with some changes, it would be possible to use this interface in mappers directly, and then it's not enough because we should also use the same EntityReference/report classes in mappers. Maybe we should do that? Have a look at the other comments and let's discuss it on Zulip.

Whatever happens, please rename the interface exposed to end users to `SearchIndexingPlanSynchronizationStrategy`, which is more consistent with the naming of `SearchIndexingPlan`. Yes, that means two interfaces with the same name, but in two mutually exclusive modules.